apis/v1alpha1/httproute_types.go

```diff
@@ -145,13 +146,23 @@ type HTTPRouteMatch struct {
 	// +optional
 	// +kubebuilder:validation:Enum=Exact;ImplementationSpecific
 	// +kubebuilder:default=Exact
-	HeaderMatchType *string `json:"headerMatchType" protobuf:"bytes,3,opt,name=headerMatchType"`
+	Type *string `json:"type" protobuf:"bytes,3,opt,name=type"`
 	// Headers are the HTTP Headers to match as interpreted via
 	// HeaderMatchType. Multiple headers are ANDed together, meaning, a request
 	// must contain all the headers specified in order to select this route.
 	//
+	// +required
+	Values map[string]string `json:"values" protobuf:"bytes,4,rep,name=values"`
+}
+
+// HTTPRouteMatch defines the predicate used to match requests to a
+// given action.
+type HTTPRouteMatch struct {
+	// +optional
+	Path *HTTPPathMatch `json:"path" protobuf:"bytes,6,opt,name=path"`
+
 	// +optional
-	Headers map[string]string `json:"headers" protobuf:"bytes,4,rep,name=headers"`
+	Header *HTTPHeaderMatch `json:"headers" protobuf:"bytes,7,opt,name=headers"`
```

Comment:
As we are making `HTTPHeaderMatch` into its own struct, are there use cases where a `HTTPRouteMatch` includes multiple types of `HTTPHeaderMatch`? i.e. Does it make sense making this into a list? For example,
```
Headers []HTTPHeaderMatch
```
While `Values` in `HTTPHeaderMatch` struct can be updated to be a single value or remain a list.

Reply:
Yeh, this is definitely worth trying out.

Reply:
I think that for now the right approach is to land #241, and then specify multiple matches for this case. It will be equivalent (though more verbose) as well as more verbose. WDYT?